### PR TITLE
Add OpenGL ES support

### PR DESCRIPTION
**Pull request by zarik5**
Tested with https://github.com/alvr-org/ALVR/tree/openxr-client/alvr/client_openxr .

### CODE REVIEW

**Comment by Ralith** [comment]
Thanks! Can you justify the decision to select GL ES automatically iff targeting Android? Does Android not support regular OpenGL? Does this require the user to adjust the extensions they enable depending on whether building for Android?

**Comment by zarik5** [comment]
OpenGL is for desktop and OpenGL ES is for embedded, they are mutually exclusive. Furthermore, the OpenXR documentation says that the `XR_KHR_opengl_es_enable` extension is tailored for android at the moment. You can use an abstraction layer over the two APIs (like the glow Rust crate) but EGL code is still platform-specific.

**Comment by Ralith** [comment]
Then there's no point trying to pretend OpenGL and OpenGL ES are the same thing. Let's expose a separate `Graphics` impl instead.

**Comment by zarik5** [comment]
> Then there's no point trying to pretend OpenGL and OpenGL ES are the same thing. Let's expose a separate Graphics impl instead.

The reason behind this is, as I said, they are mutually exclusive, and also they can share some of the wrapper code, but that's a fair point. I will make a separate Graphics implementation.

**Comment by Ralith** [comment]
> they are mutually exclusive

I don't think this is actually true. For example, [Mesa's implementation](https://docs.mesa3d.org/opengles.html) should work just fine on desktop Linux, and targeting it might be desirable for testing or portability.

**Comment by zarik5** [comment]
> I don't think this is actually true

Yeah, they are both open standards and in particular OpenGL ES is preferred as a common denominator and wrapper layer since it is mostly a subset of OpenGL (see Google ANGLE). And Mesa being opensource can do what it wants. In any case the OpenXR documentation says `XR_KHR_opengl_es_enable` is Android only. We can relax this restriction once there is really support for other non-Android platforms for this extension.

**Review by Ralith — CHANGES_REQUESTED** [review]
Let's drop the `#[cfg(not(target_os = "android")]` qualifiers this introduces, since they make this a breaking change and don't serve much purpose. Apps should already be checking extension support, and the loader will return an error on attempting to use an unsupported extension anyway.

File: openxr/src/graphics/opengles.rs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+use sys::platform::*;
+
+use crate::*;
+
+/// The OpenGL ES graphics API
+///
+/// See [`XR_KHR_opengl_es_enable`] for safety details.
+///
+/// [`XR_KHR_opengl_es_enable`]: https://www.khronos.org/registry/OpenXR/specs/1.0/html/xrspec.html#XR_KHR_opengl_es_enable
+pub enum OpenGles {}
```
[the remark this answers is not in the document]
Comment (zarik5) [reply openxr/src/graphics/opengles.rs:12]: I was conflicted on this. The proper capitalization should have been OpenGLES but the compiler warns struct identifiers with 3 or more consecutive capitalized letters. If we consider "OpenGL ES" as the original name (with a space) then `OpenGlEs` is the result after casing conversion but that looks wonky. another alternative is to keep `OpenGLES` and disable the casing warning.

[the remark this answers is not in the document]
Comment (Ralith) [reply openxr/src/graphics/opengles.rs:12]: The convention is for acronyms to be treated as words, i.e. only the first letter capitalized. This helps keep type naming style distinct from constant naming style.

[the remark this answers is not in the document]
Comment (zarik5) [reply openxr/src/graphics/opengles.rs]: My only nit is that it looks wonky. Anyway that's not that important. Rebased.

**Review by Ralith — APPROVED** [review]
Thanks!

**Pull request merged**